commands/account/auth.ts

```diff
@@ -0,0 +1,183 @@
+  loadConfig,
+  getConfigPath,
+  configFileExists,
+  updateAccountConfig,
+  writeConfig,
+  createEmptyConfigFile,
+  deleteEmptyConfigFile,
+  getConfigDefaultAccount,
+} from '@hubspot/local-dev-lib/config';
+import { logger } from '@hubspot/local-dev-lib/logger';
+import {
+  getAccessToken,
+  updateConfigWithAccessToken,
+} from '@hubspot/local-dev-lib/personalAccessKey';
+import { ENVIRONMENTS } from '@hubspot/local-dev-lib/constants/environments';
+import { toKebabCase } from '@hubspot/local-dev-lib/text';
+import { Environment } from '@hubspot/local-dev-lib/types/Config';
+import { CLIAccount_NEW } from '@hubspot/local-dev-lib/types/Accounts';
+
+import { addConfigOptions, addGlobalOptions } from '../../lib/commonOpts';
+import { handleExit } from '../../lib/process';
+import { logError } from '../../lib/errorHandlers/index';
+import { i18n } from '../../lib/lang';
+import { trackCommandUsage } from '../../lib/usageTracking';
+import { addTestingOptions } from '../../lib/commonOpts';
+import { personalAccessKeyPrompt } from '../../lib/prompts/personalAccessKeyPrompt';
+import { cliAccountNamePrompt } from '../../lib/prompts/accountNamePrompt';
+import { setAsDefaultAccountPrompt } from '../../lib/prompts/setAsDefaultAccountPrompt';
+import { EXIT_CODES } from '../../lib/enums/exitCodes';
+import { uiFeatureHighlight } from '../../lib/ui';
+import { CommonArgs, ConfigArgs } from '../../types/Yargs';
+
+const i18nKey = 'commands.account.subcommands.auth';
+
+async function createOrUpdateConfig(
+  env: Environment,
+  doesConfigExist: boolean,
+  account?: number
+): Promise<CLIAccount_NEW | null> {
+  try {
+    const { personalAccessKey } = await personalAccessKeyPrompt({
+      env,
+      account,
+    });
+    const token = await getAccessToken(personalAccessKey, env);
+    const defaultName = token.hubName ? toKebabCase(token.hubName) : null;
+
+    const name = doesConfigExist
+      ? undefined
+      : (await cliAccountNamePrompt(defaultName)).name;
+
+    const updatedConfig = await updateConfigWithAccessToken(
+      token,
+      personalAccessKey,
+      env,
+      name,
+      !doesConfigExist
+    );
+
+    if (!updatedConfig) return null;
+
+    if (doesConfigExist && !updatedConfig.name) {
+      updatedConfig.name = (await cliAccountNamePrompt(defaultName)).name;
+    }
+
+    if (doesConfigExist) {
+      updateAccountConfig({
+        ...updatedConfig,
+        environment: updatedConfig.env,
+        tokenInfo: updatedConfig.auth!.tokenInfo,
+      });
+      writeConfig();
+    }
+
+    return updatedConfig as CLIAccount_NEW;
+  } catch (e) {
+    logError(e);
+    return null;
+  }
+}
+
+export const describe = null; // i18n(`${i18nKey}.describe`);
+export const command = 'auth';
+
+type AccountInfoArgs = CommonArgs & ConfigArgs;
+
+export async function handler(
+  args: ArgumentsCamelCase<AccountInfoArgs>
+): Promise<void> {
+  const { providedAccountId, disableTracking } = args;
+
+  if (!disableTracking) {
```

Comment:
You shouldn't need this check for if tracking is disabled. trackCommandUsage calls isTrackingAllowed